Add QRcode example to docs and add dogApp

prakhargupta1 opened the pull request:
- Partial of https://github.com/mui/mui-toolpad/issues/1654

This PR is about adding the 1 example app to docs. The other 2 examples are commented out. After the 2 apps are also made live, we will link these three docs pages from the landing page.

Note: I didn't explore embedding this app in the docs as in Codesandbox it shows both preview and edit mode. Embedding would be better in the case of a deployed app.

<img width="1508" alt="Screenshot 2023-06-07 at 9 19 26 AM" src="https://github.com/mui/mui-toolpad/assets/92228082/141a4ebe-44af-46f3-986b-95fd387b8690">

prakhargupta1 commented:
I wanted to add buttons like: 
<img width=“1003” alt=“Screenshot 2022-12-13 at 11 35 20 AM” src=“https://user-images.githubusercontent.com/92228082/207239398-94e5006a-e8fa-481e-9d0c-eace385fbc33.png”>

but couldn't. In this PR I have updated some content. Need help in linking the below-mentioned apps with each use case:
1. Monitoring dashboard: https://master--toolpad.mui.com/deploy/clbf7egoc0001sg87va0oc8q1/pages/ip23ggo
2. Admin panel: https://master--toolpad.mui.com/deploy/clbff2xyw000iny9jei6me80n/pages/8n63lnb
3. BI dashboard: https://master--toolpad.mui.com/deploy/clbff2xyw000iny9jei6me80n/pages/y4d3lzj

bharatkashyap commented:
I've made the changes.

About the content, I feel that "Monitoring Dashboard" and "BI Dashboard" might not be different enough to be separate categories

prakhargupta1 commented:
@gerdadesign We are making an Examples section in Toolpad docs and want it to be linked from our landing page. As shown in one of the comments, the earlier approach was to show 3 buttons. But instead, it would be better to just add a single link.
Need your inputs on how and where we should put it. 

In the scope of this PR, we'll also change the illustrations to real app images. We can also remove the text and just write use case titles.

prakhargupta1 commented:
> But instead, it would be better to just add a single link.

@gerdadesign Let's do this later. For now, I'll just add screenshots along with 3 buttons and share here, we can then see if it needs any improvement.

prakhargupta1 commented:
@Janpot Can we include this as well in today's release?

I have excluded ChatGPT app. Things that need to be done:
1. Verify 3 apps (user-admin app needs some fixing in calling queries)
2. Add Codesandbox link

Later we can:
1. Deploy these on render.
2. Link them from landing page.

prakhargupta1 commented:
Issues that I am facing:

1. Due to https://github.com/mui/mui-toolpad/issues/1984, there are 2 pages in the project. Both have same Ids, can I delete anyone?
2. I have replaced react-vis with recharts, but it could't work until I added --viteRuntime in package.json scripts. I thought it was made default in v0.1.10
3. Preview is having some issues, the data that was visible in Edit mode wasn't visible in Preview mode:

https://github.com/mui/mui-toolpad/assets/92228082/5fddb4d5-3eba-4b53-ba35-99845f44ab92

prakhargupta1 commented:
@Janpot Currently there are 4 apps (projects) in this PR. I kept them separate as I thought it would be easier for users to understand the codebase of each app. How should we ensure there is a demo available for each app without having the need to start a new render service for each.

One way could be that we put smaller apps (custom-datagrid-column, dog-app) in codesandbox and merge the remaining ones in a single project and show them without shell in the docs.

prakhargupta1 commented:
Once merged, I'll add the showcase app to render by replacing the details [here](https://github.com/mui/mui-public/blob/36352f4d7601a90c8c572c7d464131a29b39175e/render.yaml).

Then in a separate PR I'll embed the render version of these apps in the docs.

bharatkashyap commented on docs/data/pages.ts:
```diff
@@ -125,6 +125,30 @@ const pages: MuiPage[] = [
+    pathname: '/toolpad/examples',
+    icon: DescriptionIcon,
+    children: [
+      // { pathname: '/toolpad/examples/admin-app' },
```
We can get rid of any unused paths instead of leaving them in the comments

prakhargupta1 replied:
This will come in handy soon enough. So I thought of keeping it.

bharatkashyap replied:
I think we can add it when it's needed

bharatkashyap commented on docs/src/content/useCases.js:
```diff
@@ -1,6 +1,7 @@
+import ROUTES from '../route';
 
 const useCases = {
   overline: 'Use cases',
```
To my mind, even the current description text 
<img width="1279" alt="Screenshot 2023-06-07 at 3 32 27 PM" src="https://github.com/mui/mui-toolpad/assets/19550456/9d4a06ec-d1b9-4ab4-bb1d-db971f805038"> can be too long, so we should probably aim at reducing this instead of adding more

prakhargupta1 replied:
There is no buttons yet we'll put them when we'll put all 3 apps on the docs.

bharatkashyap commented on examples/dog-app/README.md:
```diff
@@ -0,0 +1,6 @@
+<p class="description">A basic Toolpad app that shows how to call an API</p>
+
+The app can be accessed from:
+Todo: Add codesandbox link
```
Do we need to leave this reminder in the `README.md`? We can add this folder when the link is ready instead?

prakhargupta1 replied:
I added the link; it should work when this PR is merged.

bharatkashyap commented on examples/custom-datagrid-column/README.md:
```diff
@@ -1,5 +1,7 @@
-# Custom datagrid column example
+# Custom datagrid column
```
Do we need this? This is not accessible from the docs and is a duplicate of the how-to guide to customize datagrids in the new docs

prakhargupta1 replied:
As we discussed, only a selected few will be available in the docs. Can we link the how-to guide to this folder?
I have kept the dog app here only because we have mentioned it in the docs.

bharatkashyap replied:
https://deploy-preview-2082--mui-toolpad-docs.netlify.app/toolpad/how-to-guides/customize-datagrid/ we can add a link to this app from this page. We should then match the example showcased in those docs with this example app.

bharatkashyap replied:
The link to Codesandbox in the README at https://github.com/mui/mui-toolpad/tree/master/examples/custom-datagrid-column is failing for me https://codesandbox.io/p/sandbox/github/mui/mui-toolpad/tree/master/examples/custom-datagrid-column. Does this work for you @prakhargupta1?

bharatkashyap replied:
I've changed the example and it seems to work https://jt4oxx-3000.csb.app/_toolpad/app/pages/uVF80NP

prakhargupta1 replied:
This link: https://codesandbox.io/p/sandbox/github/mui/mui-toolpad/tree/master/examples/custom-datagrid-column
from production is not working.

bharatkashyap commented on docs/pages/toolpad/examples/KPI-performance.js:
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
```
We should add files only when they are being used

bharatkashyap commented on docs/pages/toolpad/examples/admin-app.js:
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import MarkdownDocs from 'docs/src/modules/components/MarkdownDocs';
```
idem

bharatkashyap commented on docs/data/pages.ts:
```diff
@@ -125,6 +125,30 @@ const pages: MuiPage[] = [
+    children: [
+      // { pathname: '/toolpad/examples/admin-app' },
+      // { pathname: '/toolpad/examples/KPI-performance' },
+      { pathname: '/toolpad/examples/qr-generator' },
```
```suggestion
      { pathname: '/toolpad/examples/qr-generator' },
      { title: 'QR Code generator' }
```

bharatkashyap commented on docs/data/pages.ts:
```diff
@@ -125,6 +125,30 @@ const pages: MuiPage[] = [
     ],
   },
+  {
+    pathname: '/toolpad/examples',
```
Based on https://github.com/mui/mui-toolpad/issues/2052#issue-1719145680, we should add "Examples" as a sub-section to the "Getting started" section

prakhargupta1 replied:
I assumed we'll do it under the restructuring PR. 
Also since we don't support 3 levels of hierarchy in the docs, does that mean putting "Examples" under "Getting started" section would mean: The 3 examples would come on the right navigation?

bharatkashyap replied:
Not sure what you mean but I've added the change in the latest commit to this PR

prakhargupta1 replied:
Looks fine, thanks.

bharatkashyap commented on examples/qr-generator/README.md:
```diff
@@ -1,5 +1,14 @@
 # QR code generator
 
-Toolpad application that defines a backend function to encode some text into a QR code.
+<p class="description">A basic Toolpad application that defines a backend function to encode some text/URL into a QR code.</p>
```
Can we add some tutorial/descriptive instructions in this? I am contrasting with this [benchmark](https://interval.com/examples/qr-codes)

bharatkashyap reviewed (changes requested):
I've added some comments regarding content/structure/conventions

bharatkashyap commented on examples/dog-app/README.md:
```diff
@@ -0,0 +1,9 @@
+
+To build this app step-by-step, visit the [docs](https://mui.com/toolpad/getting-started/quickstart/#building-your-first-application)
+
+[Open in CodeSandbox](https://codesandbox.io/p/sandbox/github/mui/mui-toolpad/tree/master/examples/dog-app)
```
Preview of the dog app example: https://3g6k84-3000.csb.app/preview/pages/PCYkw5V

prakhargupta1 replied:
This link doesn't work for me:

<img width="618" alt="Screenshot 2023-06-07 at 8 57 38 PM" src="https://github.com/mui/mui-toolpad/assets/92228082/db93ab19-8357-45bf-88cc-6529299c1192">

bharatkashyap replied:
> This link doesn't work for me:
> 
> <img alt="Screenshot 2023-06-07 at 8 57 38 PM" width="618" src="https://user-images.githubusercontent.com/92228082/244118138-db93ab19-8357-45bf-88cc-6529299c1192.png">

Interesting, I guess the codesandbox build link expires soon. This should work instead, but it will build the app when you visit it: https://codesandbox.io/p/sandbox/github/mui/mui-toolpad/tree/landing_page_add_examples/examples/dog-app

prakhargupta1 commented:
Can we merge this? @mui/toolpad

Janpot commented on docs/data/toolpad/landing/useCases.js:
```diff
@@ -1,6 +1,7 @@
 import * as React from 'react';
 import Typography from '@mui/material/Typography';
 import GradientText from 'docs/src/components/typography/GradientText';
+//  import ROUTES from '../route';
```
Let's not check in commented code, the Urls in `../route` are not valid anyway

Janpot commented on examples/custom-datagrid-column/toolpad/components/AgeColumn.tsx:
Why was the whole `custom-datagrid-column` example replaced? Can you leave it in place and add a new example instead?

prakhargupta1 replied:
Yes, it makes more sense.

Janpot replied:
The existing custom column example is showing how to use the whole row. I've linked to it in github issues. The new example doesn't showcase this and breaks old links. I think the new example should go in its own folder.

prakhargupta1 replied:
Ah got it. I'll create a new example to stay in sync with the docs.

Janpot commented:
Nit, but can we name the example after the feature instead of the dummy data? e.g. `datagrid-columns`
